src/schema.ts

```diff
@@ -76,22 +86,33 @@ export const resolvers = {
       return findUser;
     },
 
-    users: async (_, list: { numUsers: number }, context: Context): Promise<UserList> => {
+    users: async (_, list: { data: UserQueryInput }, context: Context): Promise<UserList> => {
       const secret = process.env.JWT_SECRET ?? 'secret';
       jwt.verify(context.token, secret);
 
       const DEFAULT_NUMBER_OF_USERS = 10;
-      const takeUsers = list.numUsers ?? DEFAULT_NUMBER_OF_USERS;
+      const takeUsers = list.data.numUsers ?? DEFAULT_NUMBER_OF_USERS;
+      const skipUsers = list.data.skip ?? 0;
 
       const userRepository = getRepository(User);
-      const usersList = await userRepository.find({ order: { name: 'ASC' }, take: takeUsers });
+      const response = await userRepository.findAndCount({ order: { name: 'ASC' }, skip: skipUsers, take: takeUsers });
+      const usersList = response[0];
+      const totalUsers = response[1];
+
+      const users = usersList.length;
+
+      let hasPreviousPage = true;
+      if (skipUsers === 0 || usersList.length === 0) {
+        hasPreviousPage = false;
+      }
+      const hasNextPage = skipUsers + takeUsers >= totalUsers ? false : true;
```

Comment:
Aqui tbm, vc pode retornar direto o resultado da expressão:

```suggestion
      const hasNextPage = skipUsers + takeUsers > totalUsers;
```

Outra detalhezinho é que usar o sinal de `>` é mais correto porque se a soma `skip + take` der igual ao número de usuários, não tem próxima página né?

Reply:
Nesse caso  a expressão seria `! (skipUser + takeUsers >= totalUsers)` porque hasNextPage é false se a skip somado ao número de users for igual ou maior que o total